app/javascript/flavours/glitch/features/notifications/index.js

```diff
@@ -238,6 +288,11 @@ class Notifications extends React.PureComponent {
       ? (<FilterBarContainer />)
       : null;
 
+    // if grouping is { "favourite": true, "reblog": false, "foo": true, "bar": false }
+    // then groupBy is [ "favourite", "foo" ]
+    const groupBy = grouping.reduce((acc, enabled, groupBy) => enabled ? acc.push(groupBy) : acc, ImmutableList.of());
+    const notifications = this.groupUpNotifications(this.props.notifications, groupBy);
+
```

Comment:
I don't think creating a constant which is only used once and passed to a function here is necessary.
You could create it in the function itself.
Also props could be accessed in the function instead of being passed to it.